sys/net/routing/aodvv2/routingtable.c

```diff
@@ -194,20 +194,23 @@ static void _reset_entry_if_stale(uint8_t i)
 bool routingtable_offers_improvement(struct aodvv2_routing_entry_t *rt_entry,
                                      struct node_data *node_data)
 {
-    /* Check if new info is stale */
-    if (seqnum_cmp(node_data->seqnum, rt_entry->seqnum) == -1) {
-        return false;
+    if (seqnum_cmp(node_data->seqnum, rt_entry->seqnum) == 1) {
+        /* New info is more recent */
+        return true;
     }
-    /* Check if new info is more costly */
-    if ((node_data->metric >= rt_entry->metric)
-        && !(rt_entry->state != ROUTE_STATE_INVALID)) {
-        return false;
+    if ((node_data->metric <= rt_entry->metric)
+        && (rt_entry->state != ROUTE_STATE_INVALID)) {
+        /* New info is not stale and is less costly */
+        return true;
     }
-    /* Check if new info repairs an invalid route */
-    if (!(rt_entry->state != ROUTE_STATE_INVALID)) {
-        return false;
+    if ((seqnum_cmp(node_data->seqnum, rt_entry->seqnum) == 1) &&
```

Comment:
Oh wait, the error seems to be in the Draft wording o.o Thanks for pointing this out...